core/src/main/scala/org/apache/spark/SparkConf.scala

```diff
@@ -548,23 +548,6 @@ class SparkConf(loadDefaults: Boolean) extends Cloneable with Logging with Seria
       }
     }
 
-    if (contains("spark.master") && get("spark.master").startsWith("yarn-")) {
```

Comment:
@vanzin may be interested in the parts that remove support for `yarn-client` and `yarn-cluster` masters.